### PR TITLE
Add common storage access based on Store

### PR DESCRIPTION
**Pull request by aslakknutsen**
Based on the StorageGroup and the child Store and Model
definitions a common access interface is generated:

``` go
type 'Store'Storage interface {
    'Model'() 'Model'Storage
    ...
}

func New'Store'Storage(db gorm.DB) 'Store'Storage {
    return &SQL'Store'Storage{
        'model'Storage:        New'Model'DB(db),
        ...
    }
}

type SQL'Store'Storage struct {
    'model'Storage        'Model'Storage
    ...
}
```

Usage:

``` go
var db gorm.DB
group := models.Newi'Store'Storage(db)
group.'Model'().Get(ctx, id)

```

### CODE REVIEW

File: writers.go [hunk 1]
```diff
@@ -362,6 +369,32 @@ func (w *UserTypesWriter) Execute(data *UserTypeTemplateData) error {
...
+	file, err := codegen.SourceFileFor(filename)
+	if err != nil {
+		return nil, err
+	}
+	return &StorageGroupsWriter{SourceFile: file}, nil
+}
+
+// Execute writes the code for the context types to the writer.
+func (w *StorageGroupsWriter) Execute(data *StorageGroupDefinition) error {
+	fm := make(map[string]interface{})
```
Comment (bketelsen) [inline writers.go]: are you really using all of these functions in the template?

Comment (aslakknutsen) [inline writers.go]: Ha, nope! The love of CopyPaste :+1:

Comment (aslakknutsen) [inline writers.go]: Fixed in https://github.com/goadesign/gorma/pull/105/commits/6351f7205e964adab3e8175822fafc058fd3cf1b

**Comment by bketelsen** [comment]
1) Clean up the func map
2) It needs some better high-level documentation

I like the idea a lot, saves a lot of ugly code in main/setup.  Thanks for the PR!

**Comment by aslakknutsen** [comment]
1) Done
2) You want it in 'api/doc.go'? Remove reference to StorageGroup not being used for anything and something about how it's being used.. :)

**Comment by bketelsen** [comment]
2) Yes that's perfect.

**Comment by aslakknutsen** [comment]
Reworked to generate an EntryPoint interface per Store, not one global for StorageGroup.

Some docs added: https://github.com/goadesign/goa.design/pull/26

**Comment by aslakknutsen** [comment]
There is one little snatch here tho. 

The Store entry point rely on the XStorage interface. If the user wants to add extra methods to the db access logic it would need to be added to the XDB struct. Which would require a cast on usage: Store.User().(UserDB).GetLogin()

Not sure if it's currently possible to 'append' any methods to the XStorage interface? Or if we should expose XDB in the XStore interface instead (that would of course tie the Storage interface to a specific impl which is no good.).

```go
type 'Store'Storage interface {
    'Model'() 'Model'DB
    ...
}
```

**Comment by bketelsen** [comment]
I still like this idea.  Let me ponder on some possible solutions.

**Comment by raphael** [comment]
Spring cleaning, no activity on this for 10 months :)

**Pull request closed without merging**